components/table/table-wrapper.js

```diff
@@ -574,13 +579,34 @@ export class TableWrapper extends RtlMixin(PageableMixin(SelectionMixin(LitEleme
 		this._handleTableChange();
 	}
 
-	async _handleTableChange() {
+	async _handleTableChange(mutationRecords) {
+		const updates = { count: true, classNames: true, sticky: true, syncWidths: true };
+		if (mutationRecords) {
+			for (const key in updates) updates[key] = false;
+			for (const { type, removedNodes, addedNodes, target } of mutationRecords) {
+				if (type === 'attributes') {
```

Comment:
Nit: might be worth checking that the attribute which changed is `selected`? I realize that it's the only attribute we're watching, but in case we added another watch attribute in the future, that would future-proof this logic a bit.